src/main/java/atlas/parser/Parser.java

```diff
@@ -0,0 +1,138 @@
+                } else if (isNotNumber(commandsArray[1])) {
+                    throw new AtlasException("Task number provided is not a number.");
+                } else if (commandsArray.length > 2) {
+                    throw new AtlasException("Unmarking a task as undone only requires the task number.");
+                }
+
+                int unmarkIndex = Integer.parseInt(commandsArray[1]) - 1;
+                return new UnmarkCommand(unmarkIndex);
+            case TODO:
+                commandsArray = fullCommand.split("todo ");
+                if (commandsArray.length == 1) {
+                    throw new AtlasException("The description of a todo cannot be empty.");
+                }
+
+                return new TodoCommand(commandsArray[1]);
+            case DEADLINE:
+                commandsArray = fullCommand.split("deadline | /by ");
+                if (commandsArray.length == 1) {
+                    throw new AtlasException("The description of a deadline cannot be empty.");
+                } else if (commandsArray.length < 3) {
+                    throw new AtlasException("A deadline requires a description and a date, in that order.");
+                }
+
+                return new DeadlineCommand(commandsArray[1], DateTime.formatDate(commandsArray[2]));
+            case EVENT:
+                commandsArray = fullCommand.split("event | /from | /to ");
+                if (commandsArray.length == 1) {
+                    throw new AtlasException("The description of an event cannot be empty.");
+                } else if (commandsArray.length < 4) {
+                    throw new AtlasException("An event requires a description, start and end times, in that order.");
+                }
+
+                return new EventCommand(commandsArray[1],
+                        DateTime.formatDate(commandsArray[2]),
+                        DateTime.formatDate(commandsArray[3]));
+            case DELETE:
+                if (commandsArray.length == 1) {
+                    throw new AtlasException("Deleting a task as undone requires the task number.");
+                } else if (isNotNumber(commandsArray[1])) {
+                    throw new AtlasException("Task number provided is not a number.");
+                } else if (commandsArray.length > 2) {
+                    throw new AtlasException("Deleting a task only requires the task number.");
+                }
+
+                int deleteIndex = Integer.parseInt(commandsArray[1]) - 1;
+                return new DeleteCommand(deleteIndex);
+
+            case FIND:
+                commandsArray = fullCommand.split("find ");
+                if (commandsArray.length == 1) {
+                    throw new AtlasException("Finding tasks requires the task description.");
+                } else if (commandsArray.length > 2) {
+                    throw new AtlasException("Invalid task description.");
+                }
```

Comment:
Is it much cleaner to handle all these exceptions in their corresponding classes?